### PR TITLE
[RESTEASY-2985] Resteasy Server Sent Events should not reconnect when a 204 is recieved

### PR DESCRIPTION
**Pull request by petrberan**
https://issues.redhat.com/browse/RESTEASY-2985

### CODE REVIEW

**Comment by petrberan** [comment]
@jamezp  Can I ask for a review please?

**Comment by jamezp** [comment]
@petrberan Yes, sorry. I"m going to try to get to some reviews today. I got taken for some WildFly tasks and haven't been able to process the queue.

File: resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/sse/SseEventOutputImpl.java [hunk 1]
```diff
@@ -306,43 +306,38 @@ private CompletionStage<Void> internalWriteEvent(final OutboundSseEvent event) {
 
     private BuiltResponse createResponse() {
         BuiltResponse jaxrsResponse;
-        if (state.get() == CLOSED) {
-            jaxrsResponse = (BuiltResponse) Response.noContent().build();
```
Comment (jamezp) [inline resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/sse/SseEventOutputImpl.java:310]: Why did we remove this? Return "OK" when the response is closed doesn't seem right and changes behavior quite a bit.

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/sse/SseReconnectTest.java [hunk 2]
```diff
@@ -236,4 +236,28 @@ public void testEventSourceIsOpen() throws Exception {
         }
     }
 
+    /**
+     * @tpTestDetails Tests if the SSE reconnects in case the target returns HTTP code 204 No Content.
+     * @tpInfo RESTEASY-2985
+     * @tpSince RESTEasy 6.3.0
+     */
+    public void testNoContentReconnect() throws Exception {
+        Client client = ClientBuilder.newBuilder().build();
```
Comment (jamezp) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/sse/SseReconnectTest.java:245]: Why not use try-with-resources?

**Pull request closed without merging**